src/server/tiered_storage.cc

```diff
@@ -423,11 +430,18 @@ void TieredStorage::RunOffloading(DbIndex dbid) {
       disk_stats.max_file_size)
     return;
 
-  auto cb = [this, dbid, tmp = std::string{}](PrimeIterator it) mutable {
-    TryStash(dbid, it->first.GetSlice(&tmp), &it->second);
+  string tmp;
+  auto cb = [this, dbid, &tmp](PrimeIterator it) mutable {
+    if (ShouldStash(it->second)) {
+      if (it->first.WasTouched()) {
+        it->first.SetTouched(false);
+      } else {
+        TryStash(dbid, it->first.GetSlice(&tmp), &it->second);
+      }
+    }
```

Comment:
So we stick with https://github.com/dragonflydb/dragonfly/pull/3069

Reply:
I do not think it's exactly the same like what you did. I have a task to read  https://db.in.tum.de/~leis/papers/leanstore.pdf but I think second chance algorithm https://www.geeksforgeeks.org/second-chance-or-clock-page-replacement-policy/ is what we can implement now, and it is a good baseline for the launch. Do you have objections?

Reply:
No, no objections 😌